tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/KeyValuePairListEncodingTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+public class KeyValuePairListEncodingTests
+{
+    [Theory]
+    [InlineData("key1=value1;key2=value2", new[] { "key1", "value1", "key2", "value2" })]
+    [InlineData("key1=value1;;key2=value2", new[] { "key1", "value1", "key2", "value2" })]
+    [InlineData("key1=value1;;;key2=value2", new[] { "key1", "value1", "key2", "value2" })]
+    [InlineData("key1=value1;key2=value2;key3=value3", new[] { "key1", "value1", "key2", "value2", "key3", "value3" })]
+    [InlineData("key1;key2=value2", new[] { "key1", "", "key2", "value2" })]
+    [InlineData("key1;key2;key3=value3", new[] { "key1", "", "key2", "", "key3", "value3" })]
+    [InlineData("key1;;;key3;;", new[] { "key1", "", "key3", "" })]
+    public void Parse_ValidInput_ReturnsExpectedPairs(string input, string[] expectedPairs)
+    {
+        var result = KeyValuePairListEncoding.Parse(input, ';').SelectMany(pair => new[] { pair.Name, pair.Value }).ToArray();
+        Assert.Equal(expectedPairs, result);
+    }
+
+    [Theory]
+    [InlineData(new[] { "key1", "value1", "key2", "value2" }, "key1=value1;key2=value2")]
+    [InlineData(new[] { "key1", "value1", "key2", "value2", "key3", "value3" }, "key1=value1;key2=value2;key3=value3")]
+    [InlineData(new[] { "key1", "", "key2", "value2" }, "key1;key2=value2")]
+    [InlineData(new[] { "key1", "", "key2", "", "key3", "value3" }, "key1;key2;key3=value3")]
```

Comment:
What does an empty input array produce? I assume `""` but would be good to validate that here.

Reply:
yup, added that as a test case